app/src/main/java/agency/five/tmdb/ui/viewmodel/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package agency.five.tmdb.ui.viewmodel
+
+import agency.five.tmdb.MovieRepository
+import agency.five.tmdb.ui.components.MovieItemViewState
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.get
+import org.koin.core.component.inject
+
+
+class HomeScreenViewModel() : ViewModel(), KoinComponent {
+
+    private val movieRepository: MovieRepository by inject()
```

Comment:
You can use constructor injection in a following way:
1. Remove `KoinComponent` and `private val movieRepository: MovieRepository by inject()`
2. Change class contract: `class HomeScreenViewModel(private val movieRepository: MovieRepository) : ViewModel() `
3. In `MovieModule` add parameter to VM `viewModel { HomeScreenViewModel(movieRepository = get()) }
`

Now you can remove all `koin` dependencies from VM and avoid unnecessary imports. 👍